File: dotnet/src/Microsoft.Agents.Protocol.Validation/Validators/ContentValidators/ActionContentValidator.cs
using Microsoft.Agents;

namespace Microsoft.Agents.Protocol.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for ActionContent.
/// </summary>
public class ActionContentValidator : ContentValidatorBase<ActionContent>
{
    public override ValidationResult Validate(ActionContent content, ValidationContext? context = null)
    {
        // TODO: Implement specific validation rules for ActionContent
        return ValidationResult.Success();
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/ActionContentValidator.ts
import type { ActionContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';

/**
 * Validator for ActionContent.
 */
export class ActionContentValidator extends ContentValidatorBase<ActionContent> {
  public validate(content: ActionContent, context?: ValidationContext): ValidationResult {
    // ActionContent has no strict validation requirements
    return ValidationResult.success();
  }
}
